src/sinks/prometheus/exporter.rs

```diff
@@ -304,32 +310,47 @@ impl StreamSink for PrometheusExporter {
                 metrics.map = metrics
                     .map
                     .drain(..)
-                    .map(|(MetricEntry(mut metric), is_incremental_set)| {
-                        if is_incremental_set {
-                            metric.zero();
+                    .map(|(mut entry, metadata)| {
+                        if metadata.is_incremental_set {
+                            entry.0.zero();
                         }
-                        (MetricEntry(metric), is_incremental_set)
+                        (entry, metadata)
+                    })
+                    .filter(|(_metric, metadata)| {
+                        metadata.updated_at.elapsed().as_secs() < self.config.flush_period_secs
```

Comment:
This could use an `Instant::now` computed outside of the filter and `start.duration_since(metadata.updated_at).as_secs()` here.